tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -147,13 +147,22 @@ def kill_all(self):
         self.marionette.switch_to_frame()
         self.marionette.execute_async_script("GaiaApps.killAll();")
 
-    def uninstall(self, name):
-        self.marionette.switch_to_frame()
-        data_layer = GaiaData(self.marionette)
-        data_layer.set_bool_pref('dom.mozApps.auto_confirm_uninstall', True)
-        result = self.marionette.execute_async_script('GaiaApps.uninstallWithName("%s")' % name)
-        assert (result is True), 'Failed to uninstall app: %s' % result
-        data_layer.set_bool_pref('dom.mozApps.auto_confirm_uninstall', False)
+    def install(self,manifest_url):
+        self._change_state_of_app(manifest_url, 'install')
+
+    def uninstall(self,manifest_url):
+        self._change_state_of_app(manifest_url, 'uninstall')
+
+    def install_package(self, manifest_url):
+        self._change_state_of_app(manifest_url, 'installPackage')
+
+    def _change_state_of_app(self, manifest_url, desired_action):
```

Comment:
After some investigation, `navigator.mozApps.installPackage()` is async after all. I managed to make the installation work on both regular and packaged app:
```py
    def install(self, manifest_url):
        self._change_state_of_app(manifest_url, 'install')

    def uninstall(self, manifest_url):
        self._change_state_of_app(manifest_url, 'uninstall')

    def install_package(self, manifest_url):
        self._change_state_of_app(manifest_url, 'installPackage')

    def _change_state_of_app(self, manifest_url, desired_action):
        self.marionette.switch_to_frame()
        data_layer = GaiaData(self.marionette)
        preference_action = desired_action.rstrip('Package')
        data_layer.set_bool_pref('dom.mozApps.auto_confirm_{}'.format(preference_action), True)
        result = self.marionette.execute_async_script('GaiaApps.{}("{}");'.format(desired_action, manifest_url))
        assert (result is True), 'Failed to {} app: {}'.format(desired_action, manifest_url)
        data_layer.set_bool_pref('dom.mozApps.auto_confirm_{}'.format(preference_action), False)
```